server/src/executor/mod.rs

```diff
@@ -46,6 +47,35 @@ impl Executor {
         }
     }
 
+    pub(crate) fn plan_and_execute(&self, commands: Vec<ClientCommand>) {
```

Comment:
@alanlwang1 should be right here. Basically before the executor dispatches the movement commands, this function aggregates incoming movements command for each player at a given tick, so the movement vectors get summed up and normalized, achieving any combination of movements keys.